ha/core/system_health/system_health.py

```diff
@@ -439,6 +439,9 @@ def process_event(self, healthevent: HealthEvent):
             if current_health:
                 current_health_dict = json.loads(current_health)
                 specific_info = current_health_dict["events"][0]["specific_info"]
+                # Get specific_info data from consul
+                if specific_info and specific_info.get('functional_type') and not healthevent.specific_info.get('functional_type'):
+                   healthevent.specific_info['functional_type'] = specific_info.get('functional_type')
```

Comment:
In case if POD delete event, specific_info gets updated at couple of places. Can you check if that also needs an update?

Reply:
that handle by mariyappan already in previous PR using speciific_info.update function.